service/src/main/resources/db/migration/V148__pedagogical_documents.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE IF NOT EXISTS pedagogical_document (
+                                                    id uuid PRIMARY KEY DEFAULT ext.uuid_generate_v1mc(),
+                                                    child_id uuid REFERENCES person(id) NOT NULL,
+                                                    description text,
+                                                    created timestamp with time zone DEFAULT now() NOT NULL,
+                                                    created_by uuid REFERENCES employee(id),
+                                                    updated timestamp with time zone DEFAULT now() NOT NULL,
+                                                    updated_by uuid REFERENCES employee(id)
+);
+
+CREATE TRIGGER set_timestamp BEFORE UPDATE ON pedagogical_document FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
+
+ALTER TABLE attachment
+    ADD COLUMN pedagogical_document_id UUID,
+    ADD CONSTRAINT attachment_pedagogical_document_id_fkey FOREIGN KEY (pedagogical_document_id) REFERENCES pedagogical_document (id),
+    DROP CONSTRAINT created_for_fk,
+    ADD CONSTRAINT created_for_fk CHECK (num_nonnulls(application_id, income_statement_id, message_content_id, message_draft_id, pedagogical_document_id) <= 1);
+
+CREATE INDEX idx$attachment_pedagogic_document ON attachment (pedagogical_document_id);
```

Comment:
This should probably be a partial index, since we don't need to include rows in this index where `pedagogical_document_id ` is null. The other indexes in the attachment table are full indexes, but we don't need to repeat the mistake again.

So, `WHERE pedagogical_document_id IS NOT NULL` could be added to the index.